lib/rbi/rewriters/translate_rbs_sigs.rb

```diff
@@ -0,0 +1,91 @@
+# typed: strict
+# frozen_string_literal: true
+
+module RBI
+  module Rewriters
+    # Translate all RBS signature comments to Sorbet RBI signatures
+    class TranslateRBSSigs < Visitor
+      extend T::Sig
+
+      class Error < RBI::Error; end
+
+      # @override
+      #: (Node? node) -> void
+      def visit(node)
+        return unless node
+
+        case node
+        when Tree
+          visit_all(node.nodes)
+        when AttrAccessor, AttrReader, AttrWriter
+          rbs_comments = extract_rbs_comments(node)
+          rbs_comments.each do |comment|
+            node.sigs << translate_rbs_attr_type(node, comment)
+          end
+        when Method
+          rbs_comments = extract_rbs_comments(node)
+          rbs_comments.each do |comment|
+            node.sigs << translate_rbs_method_type(node, comment)
+          end
```

Comment:
Nitpick: I think this would read better as:
```suggestion
          transform_rbs_comments(node) do |comment|
            translate_rbs_method_type(node, comment)
          end
```
if we renamed `extract_rbs_comments` to `transform_rbs_comments` and allowed it to accept a block that would get `yield`ed to inside.

Reply:
https://github.com/Shopify/rbi/pull/424#discussion_r1980156335